src/frontend/js/components/DashboardModerators/ButtonChangeRoleCta/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { messagesDashboardModerators } from '../messages';
+import { Actions, Role } from '../../../types/Enums';
+import { User } from '../../../types/User';
+import { updateUser } from '../../../data/fetchApi';
+
+export interface ButtonModeratorCtaProps {
+  user: User;
+  action: Actions;
+  onChange: () => void;
+}
+
+const ButtonChangeRoleCta = ({
+  user,
+  action,
+  onChange,
+}: ButtonModeratorCtaProps) => {
+  const message =
+    action === Actions.PROMOTE
+      ? messagesDashboardModerators.promoteModerator
+      : messagesDashboardModerators.revokeModerator;
+
+  const updateStudent = async () => {
+    // Depending on action submit the requested role
+    switch (action) {
+      case Actions.PROMOTE:
+        user.role = Role.MODERATOR;
+        break;
+      case Actions.REVOKE:
+        user.role = Role.STUDENT;
```

Comment:
I know here it's not needed but adding a `break` for each `case` statement is I think a good practice to avoid future mistake